app/Actions/Alerts/RunAlertRulesAction.php

```diff
@@ -30,14 +30,8 @@
 
 class RunAlertRulesAction
 {
-    /**
-     * @var \LibreNMS\Alert\AlertRules
-     */
-    private $rules;
-    /**
-     * @var \App\Models\Device
-     */
-    private $device;
+    private AlertRules $rules;
+    private Device $device;
 
     public function __construct(Device $device, AlertRules $rules)
```

Comment:
There is an even better way.  If you define it like this, you can remove the properties and setting them in the constructor completely.
```php
public function __construct(
    private Device $device,
    private AlertRules $rules
)
```

Reply:
My ide informs me they can be readonly too:
```php
    public function __construct(
        private readonly Device $device,
        private readonly AlertRules $rules
    ) {
    }
```

Reply:
And I tend to set readonly properties as public so:
```php
    public function __construct(
        public readonly Device $device,
        public readonly AlertRules $rules
    ) {
    }
```